duckdb_test.go

```diff
@@ -41,15 +42,38 @@ func TestOpen(t *testing.T) {
 	})
 
 	t.Run("with invalid config", func(t *testing.T) {
-		db, _ := sql.Open("duckdb", "?threads=NaN")
-		err := db.Ping()
+		_, err := sql.Open("duckdb", "?threads=NaN")
 
 		if !errors.Is(err, prepareConfigError) {
 			t.Fatal("invalid config should not be accepted")
 		}
 	})
 }
 
+func TestConnPool(t *testing.T) {
+	db := openDB(t)
+	concurrency := 4
+	// set connection pool size greater than 1
+	db.SetMaxOpenConns(concurrency)
+	defer db.Close()
+	createTable(db, t)
+	res, err := db.Exec("INSERT INTO foo VALUES ('lala', ?), ('lalo', ?)", 12345, 1234)
+	require.NoError(t, err)
+	ra, err := res.RowsAffected()
+	require.NoError(t, err)
+	require.Equal(t, int64(2), ra)
+
+	for i := 0; i < concurrency; i++ {
+		// This will create new connection to the database using the Connector.Connect method
+		conn, err := db.Conn(context.Background())
+		require.NoError(t, err)
+		rows, err := conn.QueryContext(context.Background(), "select bar from foo limit 1")
+		require.NoError(t, err)
+		require.True(t, rows.Next())
+		rows.Close()
+	}
+}
```

Comment:
Not sure if cleanup matters in test cases, but when doing `db.Conn`, you need to return it to the pool using `conn.Close()` (the name is bad, it doesn't close the conn, just returns it to pool). Don't know if it might cause the `db.Close()` call to hang until process exit.

One idea could be something like this?

```go
	db := openDB(t)
 	db.SetMaxOpenConns(2) // set connection pool size greater than 1
 	defer db.Close()
 	createTable(db, t)

	// Get two separate connections and check they're consistent
	conn1, err := db.Conn(context.Background())
	conn2, err := db.Conn(context.Background())

	res, err := conn1.Exec("INSERT INTO foo VALUES ('lala', ?), ('lalo', ?)", 12345, 1234)
 	require.NoError(t, err)
 	ra, err := res.RowsAffected()
 	require.NoError(t, err)
 	require.Equal(t, int64(2), ra)

	require.NoError(t, err)
	rows, err := conn1.QueryContext(context.Background(), "select bar from foo limit 1")
	require.NoError(t, err)
	require.True(t, rows.Next())
	rows.Close()

	require.NoError(t, err)
	rows, err := conn2.QueryContext(context.Background(), "select bar from foo limit 1")
	require.NoError(t, err)
	require.True(t, rows.Next())
	rows.Close()

	err = conn1.Close()
	require.NoError(t, err)
	err = conn2.Close()
	require.NoError(t, err)
```